test/integration/job/job_test.go

```diff
@@ -2117,6 +2139,140 @@ func BenchmarkLargeIndexedJob(b *testing.B) {
+	}{
+		"regular indexed job with failures; size=1000": {
+			nPods: 1000,
+		},
+		"job with backoffLimitPerIndex with failures; size=1000": {
+			nPods:                1000,
+			backoffLimitPerIndex: ptr.To[int32](1),
+		},
+		"regular indexed job with failures; size=10_000": {
+			nPods: 10000,
+		},
+		"job with backoffLimitPerIndex with failures; size=10_000": {
+			nPods:                10000,
+			backoffLimitPerIndex: ptr.To[int32](1),
+		},
+	}
+	mode := batchv1.IndexedCompletion
+	for name, tc := range cases {
+		b.Run(name, func(b *testing.B) {
+			enableJobBackoffLimitPerIndex := tc.backoffLimitPerIndex != nil
+			defer featuregatetesting.SetFeatureGateDuringTest(b, feature.DefaultFeatureGate, features.JobBackoffLimitPerIndex, enableJobBackoffLimitPerIndex)()
+			b.ResetTimer()
+			for n := 0; n < b.N; n++ {
+				b.StopTimer()
+				jobObj, err := createJobWithDefaults(ctx, clientSet, ns.Name, &batchv1.Job{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: fmt.Sprintf("npods-%d-%d-%v", tc.nPods, n, enableJobBackoffLimitPerIndex),
+					},
+					Spec: batchv1.JobSpec{
+						Parallelism:          ptr.To(tc.nPods),
+						Completions:          ptr.To(tc.nPods),
+						CompletionMode:       &mode,
+						BackoffLimitPerIndex: tc.backoffLimitPerIndex,
+						BackoffLimit:         ptr.To(tc.nPods),
+					},
+				})
+				if err != nil {
+					b.Fatalf("Failed to create Job: %v", err)
+				}
+				validateJobPodsStatus(ctx, b, clientSet, jobObj, podsByStatus{
+					Active: int(tc.nPods),
+					Ready:  ptr.To[int32](0),
+				})
+
+				b.StartTimer()
+				remaining := 0
+				remaining = int(tc.nPods)
+				if err := wait.ExponentialBackoff(backoff, func() (done bool, err error) {
+					if err, fail := setJobPodsPhase(ctx, clientSet, jobObj, v1.PodFailed, remaining); err != nil {
+						remaining -= fail
+						b.Logf("Transient failure failing pods: %v", err)
+						return false, nil
+					}
+					return true, nil
+				}); err != nil {
+					b.Fatalf("Could not succeed the remaining %d pods: %v", remaining, err)
+				}
+				validateJobPodsStatus(ctx, b, clientSet, jobObj, podsByStatus{
+					Active: int(tc.nPods),
+					Ready:  ptr.To[int32](0),
+					Failed: int(tc.nPods),
+				})
+
+				// Cleanup Pods and Job.
+				b.StopTimer()
+
+				// make the job succeeded
+				remaining = int(tc.nPods)
+				if err := wait.ExponentialBackoff(backoff, func() (done bool, err error) {
+					if err, succ := setJobPodsPhase(ctx, clientSet, jobObj, v1.PodSucceeded, remaining); err != nil {
+						remaining -= succ
+						b.Logf("Transient failure succeeding pods: %v", err)
+						return false, nil
+					}
+					return true, nil
+				}); err != nil {
+					b.Fatalf("Could not succeed the remaining %d pods: %v", remaining, err)
+				}
+				validateJobPodsStatus(ctx, b, clientSet, jobObj, podsByStatus{
+					Ready:     ptr.To[int32](0),
+					Succeeded: int(tc.nPods),
+					Failed:    int(tc.nPods),
+				})
+				validateJobSucceeded(ctx, b, clientSet, jobObj)
```

Comment:
why do we need to make the job succeed as part of the cleanup?

Reply:
I don't think we need, we could make the test shorter by this. It was sort of sanity test for me to verify the test is doing want. It is not measured.

Reply:
Extracted